crates/rspack_core/src/dependency/dependency_location.rs

```diff
@@ -180,9 +177,52 @@ impl SourceLocation for swc_core::common::SourceMap {
         line: hi.line,
         column: hi.col_display,
       },
-    )
+    ))
+  }
+}
+
+impl SourceLocation for &str {
+  fn look_up_range_pos(&self, start: u32, end: u32) -> Option<(SourcePosition, SourcePosition)> {
```

Comment:
I can't implement that method for dyn `rspack_sources::Source`. Perhaps we should implement a `lookup_char_pos` method on the `Source` trait, similar to the one in `swc_core::common::SourceMap`.